presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanAssert.java

```diff
@@ -36,4 +36,15 @@ public static void assertPlan(Session session, Metadata metadata, Plan actual, P
             assertTrue(matches, format("Plan does not match:\n %s\n, to pattern:\n%s", logicalPlan, pattern));
         }
     }
+
+    public static void assertNotPlan(Session session, Metadata metadata, Plan actual, PlanMatchPattern pattern)
+    {
+        requireNonNull(actual, "root is null");
+
+        boolean matches = actual.getRoot().accept(new PlanMatchingVisitor(session, metadata), new PlanMatchingContext(pattern));
+        if (matches) {
+            String logicalPlan = textLogicalPlan(actual.getRoot(), actual.getTypes(), metadata, session);
+            assertTrue(false, format("Plan matches:\n%s\nto pattern:\n%s", logicalPlan, pattern));
+        }
```

Comment:
It is a copy-paste code from above method. I would prefer extract common parts as methods.